### PR TITLE
[CORE-1169] ListJob pagination

### PR DESCRIPTION
**Pull request by armaanv**

### CODE REVIEW

File: src/server/pps/server/api_server.go [hunk 1]
```diff
@@ -760,6 +767,16 @@ func (a *apiServer) listJob(
 	jobs := a.jobs.ReadOnly(ctx)
 	jobInfo := &pps.JobInfo{}
 	_f := func(string) error {
+		if number == 0 {
+			return errutil.ErrBreak
+		}
+		if paginationMarker != nil {
+			createdAt := time.Unix(int64(jobInfo.Created.GetSeconds()), int64(jobInfo.Created.GetNanos())).UTC()
+			fromTime := time.Unix(int64(paginationMarker.GetSeconds()), int64(paginationMarker.GetNanos())).UTC()
+			if createdAt.Equal(fromTime) || !reverse && createdAt.After(fromTime) || reverse && createdAt.Before(fromTime) {
```
Comment (acohen4) [inline src/server/pps/server/api_server.go:776]: so we don't want to include records that were created at `fromTime`?

File: src/server/pps/server/api_server.go [hunk 2]
```diff
@@ -760,6 +767,16 @@ func (a *apiServer) listJob(
...
+				return nil
```
Comment (acohen4) [inline src/server/pps/server/api_server.go:777]: should this return errutil.ErrBreak?

File: src/server/pps/server/api_server.go (hunk 1 above)
Comment (acohen4) [reply src/server/pps/server/api_server.go:776]: nvm that makes sense based on the usage

**Review by acohen4 — APPROVED** [review]
LGTM

File: src/server/pps/server/api_server.go (hunk 1 above)
Comment (armaanv) [reply src/server/pps/server/api_server.go:776]: yeah, the usage pattern is that users will get the first page of results and then request the next page by specifying the paginationTimestamp as the createdAt time of the last job in the first page. We don't want to return that last job again so we exclude matching timestamps

File: src/server/pps/server/api_server.go (hunk 2 above)
Comment (armaanv) [reply src/server/pps/server/api_server.go:777]: hmm, I think you're right. errutil.ErrBreak will probably save some iterations over records that we know don't need to be included

**Pull request merged**